### DISCUSSION TITLE
Is it possible to change the opacity of Neovim?

### DISCUSSION THREAD

**Discussion by JulienLecoq** [neovim/neovim#30588] [Q&A]
I'm able to make it fully transparent but I would like to set it with something like 0.8 of opacity so that I can see the background of my terminal emulator and also be able to read the text inside Neovim easily.

**Comment by zeertzjq** [comment]
This is a terminal emulator configuration problem. Nvim doesn't know how your terminal emulator handles transparency.

**Comment by JulienLecoq** [comment]
Mhh, I’m not sure to understand. You mean that my terminal emulator should be the one responsible of the transparency of programs which runs inside it? So there should be a way to set the transparency of programs but directly at the terminal emulator config?

**Comment by JulienLecoq** [comment]
Because I don’t want to set the terminal transparent, I want it to be fully opaque. I have a background set up on my terminal emulator that I would like to be able to see from inside Neovim.

**Comment by sammyhori** [comment]
**Neovim doesn't natively support changing the opacity of the background colour**, it just supports turning it off.

If the issue reading text applies beyond neovim (i.e. it's also hard to read in the shell), you could add a somewhat opaque background across your whole terminal.
Some terminals support this natively, by adding a background colour which will apply over the background image and making that colour opaque. Alternatively you could edit your background image and overlay an image of your desired colour over the top of it at a opacity which suits you. You could also achieve a similar, less customisable effect by reducing the exposure of the image.

For nvim only opaqueness in a hacky manner, you could set the background opacity or image in your terminal every time you open and close nvim.

Another idea that may or may not work would be to install a terminal emulator that supports RGBA values and set the background colour to something transparent.

**Comment by wookayin** [comment]
`:hi Normal guibg=NONE`

**Reply by sammyhori** [reply]
> I would like to set it with something like 0.8 of opacity

This sets the background to transparent, not to partially opaque.

**Comment by oceanthunder** [comment]


**Reply by sammyhori** [reply]
> I don’t want to set the terminal transparent, I want it to be fully opaque. I have a background set up on my terminal emulator that I would like to be able to see from inside Neovim.

What OP wants is:
- Keep their terminal fully opaque (i.e. no transparency),
- Have a single colour background for neovim (which is supported) but they want to be able to make this neovim background partially transparent (i.e. change the opacity) (which isn't supported).
